src/data/queries/collapsiblePanel.ts

```diff
@@ -0,0 +1,21 @@
+import { ParagraphCollapsiblePanel } from '@/types/drupal/paragraph'
+import { QueryFormatter } from 'next-drupal-query'
+import { CollapsiblePanel } from '@/types/formatted/collapsiblePanel'
+import { formatParagraph } from '@/lib/drupal/paragraphs'
+
+export const formatter: QueryFormatter<
+  ParagraphCollapsiblePanel,
+  CollapsiblePanel
+> = (entity: ParagraphCollapsiblePanel) => {
+  return {
+    type: entity.type as CollapsiblePanel['type'],
```

Comment:
similar question around the type coercion for all paragraphs. Also I wonder if it makes sense to do a similar `...baseParagraphFields(entity)` for paragraph formatters, or if that's overengineering for the sake of it?

Reply:
Same answer as above (you'll see this pattern in many places). Regarding the base paragraph fields, a utility could probably be useful. I didn't find it strictly necessary given the number of fields, but I'm not ready to call it overengineering. It'd save a few keystrokes _and_ align better with how nodes are formatted. On the other side, though, the function would probably have some funkiness when trying to assign the `type` property. Not sure how funky, but it wouldn't just be `type: entity.type` because we'd need a map of some sort to do the assertion.